packages/core/types/client.d.ts

```diff
@@ -6,15 +6,98 @@ import {
   OnSessionCallback,
```

Comment:
On the question of public versus internal interfaces for the notifiers themselves we can address that later. However I don't think it makes much sense at all for `@bugsnag/core` to have this concept. We don't document/expect our customers to use `@bugsnag/core` directly anyway so we can just keep things simple and export the actual interfaces.

For now though, we're still using the manual types - we can't switch to generated types until all the JS files that make up the external interfaces are converted to TypeScript otherwise invalid types are generated.